sdk/core/azure-core/test/ut/transport_adapter_curl.cpp

```diff
@@ -37,78 +37,83 @@ namespace Azure { namespace Core { namespace Test {
-
-    auto threadRoutine = [&]() {
-      using namespace std::chrono_literals;
-      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
-      auto response = m_pipeline->Send(Azure::Core::GetApplicationContext(), request);
-      checkResponseCode(response->GetStatusCode());
-      auto expectedResponseBodySize = std::stoull(response->GetHeaders().at("content-length"));
-      CheckBodyFromBuffer(*response, expectedResponseBodySize);
-      std::this_thread::sleep_for(1s);
-    };
-
-    std::thread t1(threadRoutine);
-    std::thread t2(threadRoutine);
-    t1.join();
-    t2.join();
-
-    // 2 connections must be available at this point
-    EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 2);
-
-    std::thread t3(threadRoutine);
-    std::thread t4(threadRoutine);
-    std::thread t5(threadRoutine);
-    t3.join();
-    t4.join();
-    t5.join();
-
-    // Two connections re-used plus one connection created
-    EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 3);
-
-#ifdef RUN_LONG_UNIT_TESTS
-    {
-      // Test pool clean routine
-      std::cout << "Running Connection Pool Cleaner Test. This test takes more than 3 minutes to "
-                   "complete."
-                << std::endl
-                << "Add compiler option -DRUN_LONG_UNIT_TESTS=OFF when building if you want to "
-                   "skip this test."
-                << std::endl;
-
-      // Wait for 180 secs to make sure any previous connection is removed by the cleaner
-      std::this_thread::sleep_for(std::chrono::milliseconds(1000 * 180));
-
-      std::cout << "First wait time done. Validating state." << std::endl;
-
-      // index is not affected by cleaner. It does not remove index
-      EXPECT_EQ(Http::CurlConnectionPool::ConnectionsIndexOnPool(), 1);
-      // cleaner should have removed connections
-      EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 0);
-
-      std::thread t1(threadRoutine);
-      std::thread t2(threadRoutine);
-      t1.join();
-      t2.join();
-
-      // wait for connection to be moved back to pool
-      std::this_thread::sleep_for(std::chrono::milliseconds(1000));
-
-      // 2 connections must be available at this point and one index
-      EXPECT_EQ(Http::CurlConnectionPool::ConnectionsIndexOnPool(), 1);
-      // Depending on how fast the previous requests are sent, there could be one or more
-      // connections in the pool. If first request is too fast, the second request will reuse the
-      // same connection.
-      EXPECT_PRED1(
-          [](int currentConnections) { return currentConnections > 1; },
-          Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"));
-    }
-#endif
-  }
+  // commenting test since it can't be disabled as it depends on a friend declaration to access
+  // private fields
+  //   TEST_P(TransportAdapter, connectionPoolTest)
```

Comment:
How about surrounding it using `#if` instead of commenting it out:
https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-large-comments

```C++
#if TEST_DISABLED
  test code
#endif
```